thunder/tests/distributed/test_ddp.py

```diff
@@ -1021,6 +1037,169 @@ def fwd_loss(m, x):
         with thunder.ThunderModule.no_sync(model):
             fwd_loss(model, x)
 
+    @pytest.mark.skipif(torch.cuda.device_count() < 2, reason="")
+    @common_utils.parametrize("name", ("column", "row"))
+    def test_tensor_parallel_linear(self, name):
+        device = torch.device("cuda", self.rank)
+        x = torch.randn(2, 12).to(device).requires_grad_()
+
+        process_group = None
+        ref_model = ToyModel().to(device)
+        with c10d._coalescing_manager(async_ops=True) as cm:
```

Comment:
for my own edification: why are we all_reducing here?

Reply:
all the ranks have the same initial weights, which I don't think mandatory for the purpose